### PR TITLE
*: use `if (container.contains(value))` instead of `if (container.count(value))`

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #6233 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by ywqzzy** [comment]
Why not using bitset?

**Comment by Lloyd-Pottiger** [comment]
> Why not using bitset?

bitset can only be fixed size. `std::bitset<n> b` where n must be a known constant integer.

File: dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h [hunk 1]
```diff
@@ -307,7 +292,7 @@ class DMFilePackFilter
     RSCheckParam param;
 
     std::vector<RSResult> handle_res;
-    std::vector<UInt8> use_packs;
+    std::vector<bool> use_packs;
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h:295]: I think `boost::dynamic_bitset` is a better choice when you really need it.

> If the size of the bitset is known at compile time, [std::bitset](https://en.cppreference.com/w/cpp/utility/bitset) may be used, which offers a richer set of member functions. In addition, [boost::dynamic_bitset](https://www.boost.org/doc/libs/release/libs/dynamic_bitset/dynamic_bitset.html) exists as an alternative to std::vector<bool>.

Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h:295]:
```cpp
namespace boost {

template <typename Block, typename Allocator>
class dynamic_bitset
{
public:

    class [reference](https://www.boost.org/doc/libs/1_36_0/libs/dynamic_bitset/dynamic_bitset.html#reference)
    {
        void operator&(); // not defined
    public:
        // An automatically generated copy constructor.

        reference& operator=(bool value);
        reference& operator=(const reference& rhs);

        reference& operator|=(bool value);
        reference& operator&=(bool value);
        reference& operator^=(bool value);
        reference& operator-=(bool value);

        bool operator~() const;
        operator bool() const;
        reference& flip();
    };
    typedef bool [const_reference](https://www.boost.org/doc/libs/1_36_0/libs/dynamic_bitset/dynamic_bitset.html#const_reference);


    reference [operator[]](https://www.boost.org/doc/libs/1_36_0/libs/dynamic_bitset/dynamic_bitset.html#bracket)(size_type pos);
    bool [operator[]](https://www.boost.org/doc/libs/1_36_0/libs/dynamic_bitset/dynamic_bitset.html#const-bracket)(size_type pos) const;
};
```

seems that dynamic_bitset operator[] also return the reference 😭

Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h:295]: https://www.boost.org/doc/libs/1_36_0/libs/dynamic_bitset/dynamic_bitset.html

**Review by lidezhu — APPROVED** [review]
LGTM

**Comment by lidezhu** [comment]
By the way, the pr's title may need be changed?

File: dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h (hunk 1 above)
Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h]: @JaySon-Huang `dynamic_bitset` make iter inconvenient, so keep `vector<UInt8>`.

**Review by hongyunyan — APPROVED** [review]

File: dbms/src/Storages/DeltaMerge/StableValueSpace.cpp [hunk 2]
```diff
@@ -357,23 +357,13 @@ void StableValueSpace::calculateStableProperty(const DMContext & context, const
...
-                    use_packs_count += 1;
-            }
-            if (unlikely((size_t)new_pack_properties.property_size() != use_packs_count))
-            {
-                throw Exception(
-                    fmt::format("size doesn't match [new_pack_properties_size={}] [use_packs_size={}]", new_pack_properties.property_size(), use_packs_count),
-                    ErrorCodes::LOGICAL_ERROR);
-            }
+            const size_t use_packs_count = std::count(use_packs.begin(), use_packs.end(), true);
+            RUNTIME_CHECK_MSG(static_cast<size_t>(new_pack_properties.property_size()) == use_packs_count, "size doesn't match [new_pack_properties_size={}] [use_packs_size={}]", new_pack_properties.property_size(), use_packs_count);
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/StableValueSpace.cpp:364]:
```suggestion
            RUNTIME_CHECK_MSG(static_cast<size_t>(new_pack_properties.property_size()) == use_packs_count, "size doesn't match, new_pack_properties_size={} use_packs_size={}", new_pack_properties.property_size(), use_packs_count);
```

**Review by JaySon-Huang — APPROVED** [review]
LGTM with small comment

**Pull request merged**